torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -2060,6 +2084,42 @@ Tensor* TensorExprKernel::computeConv2d(const torch::jit::Value* v) {
   return new Tensor(ResultBuf.node(), s);
 }
 
+Tensor* TensorExprKernel::computeMatmul(const torch::jit::Value* v) {
+  const Node* n = v->node();
+  auto const& shape = sizesForValue(v);
+  Dtype dtype = kFloat;
+  auto maybe_stype = findDtypeForValue(v);
+  if (maybe_stype) {
+    dtype = Dtype(*maybe_stype);
+  }
+  BufHandle ResultBuf("matmul", shape, dtype);
+  const Buf* a = tensors_.at(n->input(0))->buf();
+  const Buf* b = tensors_.at(n->input(1))->buf();
+
+  auto size_a = ExprVectorToExprHandleVector(a->dims());
+  auto size_b = ExprVectorToExprHandleVector(b->dims());
+  const IntImm* total_size = dynamic_cast<const IntImm*>(
+      IRSimplifier::simplify((size_a[0] * size_a[1] * size_b[1])).node());
```

Comment:
Should we be multiplying by `size_b[0]` as well ?

Reply:
It's a rough estimate of amount of work matmul needs to do, which is N*M*K (size_a[1] == size_b[0] == M). Would I not pass an interview with such analysis? 😜